java-client/src/test/java/org/opensearch/client/opensearch/integTest/aws/AwsSdk2TransportMock.java

```diff
@@ -0,0 +1,46 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.client.opensearch.integTest.aws;
+
+import org.opensearch.client.transport.Endpoint;
+import org.opensearch.client.transport.TransportOptions;
+import org.opensearch.client.transport.aws.AwsSdk2Transport;
+import org.opensearch.client.transport.aws.AwsSdk2TransportOptions;
+import org.opensearch.client.util.OpenSearchRequestBodyBuffer;
+import software.amazon.awssdk.http.SdkHttpClient;
+import software.amazon.awssdk.http.SdkHttpFullRequest;
+import software.amazon.awssdk.regions.Region;
+
+import java.io.IOException;
+
+import javax.annotation.CheckForNull;
+import javax.annotation.Nonnull;
+
+public class AwsSdk2TransportMock extends AwsSdk2Transport {
```

Comment:
I don't think that "mock" is correct naming here. That implies something that "fakes" a certain operation, which this does not.